presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -583,12 +605,18 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
         AggregationNode aggregationNode = new AggregationNode(
                 idAllocator.getNextId(),
                 subPlan.getRoot(),
-                aggregations.build(),
+                aggregations,
                 groupingSymbols,
                 AggregationNode.Step.SINGLE,
                 Optional.empty(),
                 groupIdSymbol);
 
+        if (aggregationNode.hasEmptyGroupingSet() && aggregationNode.hasNonEmptyGroupingSet() && !aggregationNode.getOrderBySymbols().isEmpty()) {
+            throw SemanticExceptions.notSupportedException(
```

Comment:
Could you add a comment why this isn't supported? Is this because of @pnowojski improvement?